src/jvm/code/method_body.rs

```diff
@@ -132,6 +138,15 @@ impl<I> InstructionList<I> {
             .map(|(k, _)| *k)
     }
 
+    /// Returns the program counter of the previous instruction before the given one.
+    #[must_use]
+    pub fn prev_pc_of(&self, pc: &ProgramCounter) -> Option<ProgramCounter> {
```

Comment:
Do you have some application scenarios of this function in mind? For example, the reason why `next_pc_of` exists is that, when doing static analysis, we need to know the next instruction to execute after the current one is executed.

It would be nice to know your motivation for adding this function.

Reply:
In one of my projects, I am using the SSA IR format to generate a custom bytecode blob of each method, that is run on a custom mini VM. When performing the static analysis on some parts of the method, I am iterating forward through the blob, and sometimes I require a backtrack, to either optimize out some instructions (when lifting to my own format), or to search back from the current point.

I also plan on using this in conjunction with the `exit_point` (possible name change needed that you brought up in the last comment) function, to move backwards in the IR. 😄

Reply:
Okay, this is a reasonable scenario. I'm thinking, if it's possible to use the `InstructionList.iter().rev()` method as you mentioned you're iterating over, that would be a good option.

Reply:
This would, however in my current implementations of some passes, I jump around quite a bit, so I think this function would still be benificial to include.

Reply:
Make sense to me.